[aten] remove warning on change of sign

jspark1105 opened the pull request:
Summary:
Remove warning
```
caffe2/aten/src/ATen/native/cuda/Reduce.cuh(654): warning: integer conversion resulted in a change of sign
```
When acc_ptr_ != nullptr , numerator_ and denominator_ must have been initialized.
Other minor changes:
Make member variables of AccumulationBuffer private
size_factor_ is used nowhere.

Test Plan: CI

Differential Revision: D20181169

ezyang reviewed (changes requested):
In this case, the warning seems to properly be warning us about something very fishy here. numerator and denominator are stored as `size_t`, which means that -1 is going to become 0xFFFFFFFF. This seem incredibly unlikely to be the intended behavior. Can you dig deeper and figure out what the right fix is?

ezyang commented:
Alternately, find the person who added these -1s and make them investigate.

jspark1105 committed:
[aten] remove warning on change of sign (#34016)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/34016

Remove warning
```
caffe2/aten/src/ATen/native/cuda/Reduce.cuh(654): warning: integer conversion resulted in a change of sign
```

When acc_ptr_ != nullptr , numerator_ and denominator_ must have been initialized.

Other minor changes:
* Make member variables of AccumulationBuffer private
* size_factor_ is used nowhere.

Test Plan: CI

Reviewed By: dskhudia

Differential Revision: D20181169

fbshipit-source-id: b4f267810507ab22974aa5af7fee016a18dcb274
